test/integration/controller/core/queuedworkload_controller_test.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"fmt"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	kueue "sigs.k8s.io/kueue/api/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	"sigs.k8s.io/kueue/test/integration/framework"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("QueuedWorkload controller", func() {
+	var (
+		ns              *corev1.Namespace
+		queue           *kueue.Queue
+		badQueue        *kueue.Queue
+		nonCreatedQueue *kueue.Queue
+		clusterQueue    *kueue.ClusterQueue
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "core-queuedworkload-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.BeforeEach(func() {
+		clusterQueue = testing.MakeClusterQueue("cq-workload").
+			Resource(testing.MakeResource(resourceGPU).
+				Flavor(testing.MakeFlavor(flavorModelA, "5").Ceiling("10").Obj()).
+				Flavor(testing.MakeFlavor(flavorModelB, "5").Ceiling("10").Obj()).Obj()).Obj()
+		gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+		queue = testing.MakeQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, queue)).To(gomega.Succeed())
+		badQueue = testing.MakeQueue("badqueue", ns.Name).ClusterQueue("badCL").Obj()
+		gomega.Expect(k8sClient.Create(ctx, badQueue)).To(gomega.Succeed())
+		nonCreatedQueue = testing.MakeQueue("nonCreatedqueue", ns.Name).Obj()
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(framework.DeleteQueue(ctx, k8sClient, queue)).To(gomega.Succeed())
+		gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQueue)).To(gomega.Succeed())
+	})
+
+	ginkgo.It("Should update status when workloads are created", func() {
+		cases := []struct {
```

Comment:
I'm fine with this for now, but it's not really an encouraged pattern for gomega.

This is the pattern https://onsi.github.io/ginkgo/#separating-creation-and-configuration-justbeforeeach

Which is one of the reasons I don't really like gomega (it just complicates things :joy:)

Reply:
dear god, I get you, how complicated, test_cases are so much simpler

Reply:
happy to address the test comments on a follow up PR, this PR is getting too big.

Reply:
yes

Reply:
decided to do it any way :)